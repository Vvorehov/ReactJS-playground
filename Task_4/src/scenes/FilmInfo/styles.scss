$brandColor: #ED5664;

.film-info {
  color: #fff;
  padding: 2rem 0;
}

.film-info__poster {
  display: block;
  width: 100%;
}

.film-info__details {
  padding-top: 2rem;
}

.film-info__name {
  display: inline-block;
  font-size: 3.2rem;
  font-weight: 700;
  padding-right: 5rem;
  position: relative;
  margin-bottom: 1rem;
  color: $brandColor;
}

.film-info__rating {
  display: block;
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  right: 0;
  font-size: 2.2rem;
  font-weight: 300;
  text-align: center;
  height: 4rem;
  width: 4rem;
  line-height: 3.6rem;
  vertical-align: middle;
  color: green;
  border: 1px solid green;
  border-radius: 50%;
}

.film-info__genre {
  font-size: 1.4rem;
  margin-bottom: 2rem;
}

.film-info__info {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  margin-bottom: 3rem;
  font-size: 1.4rem;
  font-weight: 700;
  span {
    margin-right: 2rem;
  }
}

.film-info__desc {
  font-size: 1.4rem;
  line-height: 1.4;
}