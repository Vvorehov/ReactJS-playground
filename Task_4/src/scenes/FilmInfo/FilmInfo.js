import React, { Component } from "react";

import "./styles.scss";

class FilmInfo extends Component {

  render() {
    let genres = this.props.genres ? this.props.genres.join(", ") : "";
    return (
      <div className="film-info">
        <div className="row">
          <div className="col-sm-4">
            <img src={this.props.poster_path} alt={this.props.title} className="film-info__poster"/>
          </div>
          <div className="col-sm-8">
            <div className="film-info__details">
              <h1 className="film-info__name">
                {this.props.title}
                <span className="film-info__rating">{this.props.vote_average}</span>
              </h1>
              <p className="film-info__genre">{genres}</p>
              <p className="film-info__info">
                <span>{(new Date(this.props.release_date)).getFullYear()}</span>
                <span>{this.props.runtime} min</span>
              </p>
              <p className="film-info__desc">{this.props.overview}</p>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default FilmInfo;