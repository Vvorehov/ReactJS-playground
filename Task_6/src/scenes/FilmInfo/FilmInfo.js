import React, { Component } from "react";
import "./styles.scss";

class FilmInfo extends Component {

  render() {
    if (this.props.film && this.props.film.genres) {
      let genres = this.props.film.genres ? this.props.film.genres.join(", ") : "";
      return (
        <div className="film-info">
          <div className="row">
            <div className="col-sm-4">
              <img src={this.props.film.poster_path} alt={this.props.film.title} className="film-info__poster"/>
            </div>
            <div className="col-sm-8">
              <div className="film-info__details">
                <h1 className="film-info__name">
                  {this.props.film.title}
                  <span className="film-info__rating">{this.props.film.vote_average}</span>
                </h1>
                <p className="film-info__genre">{genres}</p>
                <p className="film-info__info">
                  <span>{(new Date(this.props.film.release_date)).getFullYear()}</span>
                  <span>{this.props.film.runtime} min</span>
                </p>
                <p className="film-info__desc">{this.props.film.overview}</p>
              </div>
            </div>
          </div>
        </div>
      )
    } else {
        return <h2 className="film-info_empty">Loading...</h2>
    }
  }
}

export default FilmInfo;