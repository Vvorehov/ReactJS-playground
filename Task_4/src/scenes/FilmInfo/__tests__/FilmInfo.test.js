import React from "react";
import FilmInfo from "../FilmInfo";
import TestRenderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Setup enzyme's react adapter
configure({ adapter: new Adapter() });

const testFilmInfo = {
  "id": 284054,
  "title": "Black Panther",
  "tagline": "Long live the king",
  "vote_average": 7.3,
  "vote_count": 3788,
  "release_date": "2018-02-13",
  "poster_path": "https://image.tmdb.org/t/p/w500/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
  "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan \"special forces\"), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
  "budget": 200000000,
  "revenue": 1245257672,
  "genres": [
    "Action",
    "Adventure",
    "Fantasy",
    "Science Fiction"
  ],
  "runtime": 134
};

describe('FilmInfo component', () => {
  it('shallow wrapper instance should not be null', () => {
    const wrapper = shallow(<FilmInfo {...testFilmInfo} />);
    const instance = wrapper.instance();

    expect(instance).not.toBe(null);
  });
});

describe('FilmInfo layout complete', () => {
    beforeEach(function() {

        global.fetch = jest.fn().mockImplementation(() => {
            let p = new Promise((resolve, reject) => {
                resolve({
                    ok: true,
                    Id: '284054',
                    json: function() {
                        return {Id: '284054'}
                    }
                });
            });

            return p;
        });

    });

    it('Render correct', () => {

        const component = TestRenderer.create(
            <FilmInfo {...testFilmInfo} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

