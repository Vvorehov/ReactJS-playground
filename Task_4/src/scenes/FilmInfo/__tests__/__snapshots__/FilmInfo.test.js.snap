// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmInfo layout complete Render correct 1`] = `
<div
  className="film-info"
>
  <div
    className="row"
  >
    <div
      className="col-sm-4"
    >
      <img
        alt="Black Panther"
        className="film-info__poster"
        src="https://image.tmdb.org/t/p/w500/uxzzxijgPIY7slzFvMotPv8wjKA.jpg"
      />
    </div>
    <div
      className="col-sm-8"
    >
      <div
        className="film-info__details"
      >
        <h1
          className="film-info__name"
        >
          Black Panther
          <span
            className="film-info__rating"
          >
            7.3
          </span>
        </h1>
        <p
          className="film-info__genre"
        >
          Action, Adventure, Fantasy, Science Fiction
        </p>
        <p
          className="film-info__info"
        >
          <span>
            2018
          </span>
          <span>
            134
             min
          </span>
        </p>
        <p
          className="film-info__desc"
        >
          King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan "special forces"), and an American secret agent, to prevent Wakanda from being dragged into a world war.
        </p>
      </div>
    </div>
  </div>
</div>
`;
